spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -1498,6 +1500,73 @@ def name_id_version(format_urn)
       end
     end
 
+    context 'SAML request is signed' do
+      let(:user) { create(:user, :fully_registered) }
+      let(:authn_requests_signed) { true }
+      let(:service_provider) do
+        create(
+          :service_provider,
+          certs: ['sp_sinatra_demo', 'saml_test_sp'],
+          active: true,
+          assertion_consumer_logout_service_url: 'https://example.com',
+        )
+      end
+
+      let(:auth_settings) do
+        saml_settings(
+          overrides: {
+            authn_context: [
+              Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF,
+              Saml::Idp::Constants::DEFAULT_AAL_AUTHN_CONTEXT_CLASSREF,
+            ],
+            certificate: File.read(Rails.root.join('certs', 'sp', 'saml_test_sp2.crt')),
+            issuer: service_provider.issuer,
+            private_key: OpenSSL::PKey::RSA.new(
+              File.read(Rails.root + 'keys/saml_test_sp2.key'),
+            ).to_pem,
+            security: {
+              authn_requests_signed:,
+            },
+          },
+        )
+      end
+
+      before do
+        stub_analytics
+      end
+
+      it 'notes that in the analytics event' do
+        IdentityLinker.new(user, service_provider).link_identity
```

Comment:
Mayyyybe look into updating generate_saml_response to do this "and they have a verified email" dance